src/averageAges.js

```diff
@@ -59,6 +82,37 @@ function calculateAverageAgeDiff(people, onlyWithSon) {
   // 2. keep people who have mothers in the array
   // 3. calculate the difference child.born - mother.born
   // 4. return the average value
+  const peopleWithMother = onlyWithSon
+    ? people.filter((person) =>
+      person.mother !== null && person.sex === 'm'
+    )
+    : people.filter((person) =>
+      person.mother !== null
+    );
+
+  const women = people
+    .filter((person) => person.sex === 'f')
+    .map(person => person.name);
+
+  const peopleWithExistMother = peopleWithMother
+    .filter((person) => women.includes(person.mother));
+
+  const averageAge = peopleWithExistMother
+    .reduce((prev, person) => {
+      const mother = people.find((personsMother) =>
+        personsMother.name === person.mother);
+
+      return prev + (person.born - mother.born);
+    }, 0) / peopleWithExistMother.length;
+
+  return averageAge;
+}
+
+function getAverageAge(sortedPeople) {
+  return sortedPeople
+    .reduce((prev, { born, died }) => {
+      return prev + died - born;
```

Comment:
You are calculating average age in every function separately. Instead, you could write a function that calculates average from array of numbers, and use it in all three functions. This would make your code DRY (Don't Repeat Yourself).

Reply:
I created and used it in the first 2 functions 'calculateMenAverageAge' and 'calculateWomenAverageAge'. I didn't use it in the 'calculateAverageAgeDiff' because I have a different condition and therefore a different way to calculate the average age.